src/reason-parser/reason_pprint_ast.ml

```diff
@@ -6640,6 +6644,159 @@ class printer  ()= object(self:'self)
     match xs with
     | [x] -> x
     | xs -> makeBreakableList xs
+
+  method formatFunAppl ~jsxAttrs ~args ~funExpr () =
+    (* If there was a JSX attribute BUT JSX component wasn't detected,
+       that JSX attribute needs to be pretty printed so it doesn't get
+       lost *)
+    let maybeJSXAttr = List.map self#attribute jsxAttrs in
+
+    let categorizeFunApplArgs args =
+      let reverseArgs = List.rev args in
+      match reverseArgs with
+      | (Nolabel, {pexp_desc = Pexp_fun _})::_ -> `LastArgIsCallback reverseArgs
+      | _ -> `NormalFunAppl args
+    in
+    begin match categorizeFunApplArgs args with
+    | `LastArgIsCallback reverseArgs ->
+        (* This is the following case:
+         * Thing.map(foo, bar, baz, (abc, z) =>
+         *   MyModuleBlah.toList(argument)
+         *)
+        let (args, callbackArg) = begin match reverseArgs with
```

Comment:
If you store the last arg and remaining args at the time you create`LastArgIsCallback`, then you already have them split and can remove this `raise`.